pulsar-broker-common/src/main/java/org/apache/pulsar/broker/authorization/PulsarAuthorizationProvider.java

```diff
@@ -263,7 +263,10 @@ public CompletableFuture<Void> revokePermissionAsync(TopicName topicName, String
                         policies.auth_policies.getTopicAuthentication()
                                 .computeIfPresent(topicName.toString(), (k, v) -> {
                                         v.remove(role);
-                                        return null;
+                                        if (v.isEmpty()) {
+                                            return  null;
+                                        }
+                                        return v;
```

Comment:
Was this regression caught by one of the tests? If not, we should add a test to make sure this can't happen again.

Reply:
I have added some tests for this. you can see the modification in `AuthorizationProducerConsumerTest.java`